File: csharp/1252.cells-with-odd-values-in-a-matrix.cs
/*
 * @lc app=leetcode id=1252 lang=csharp
 *
 * [1252] Cells with Odd Values in a Matrix
 */

// @lc code=start
public class Solution {
    public int OddCells(int n, int m, int[][] indices) {
        long r = 0, c = 0;
        int rc = 0, cc = 0;
        foreach (var i in indices) {
            r ^= 1L << i[0];
            c ^= 1L << i[1];
        }
        while (r != 0) {
            rc += (int)(r & 1);
            r >>= 1;
        }
        while (c != 0) {
            cc += (int)(c & 1);
            c >>= 1;
        }
        return rc * m + cc * n - 2 * rc * cc;
    }
}
// @lc code=end

File: java/1252.cells-with-odd-values-in-a-matrix.java
/*
 * @lc app=leetcode id=1252 lang=java
 *
 * [1252] Cells with Odd Values in a Matrix
 */

// @lc code=start
class Solution {
    public int oddCells(int m, int n, int[][] indices) {
        long rows = 0, cols = 0;
        for (var i : indices) {
            rows ^= (1L << i[0]);
            cols ^= (1L << i[1]);
        }
        int oddRows = 0, oddCols = 0;
        while (rows != 0) {
            oddRows += (rows & 1);
            rows >>= 1;
        }
        while (cols != 0) {
            oddCols += (cols & 1);
            cols >>= 1;
        }
        return oddRows * (n - oddCols) + oddCols * (m - oddRows);
    }
}
// @lc code=end

